sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExporterBuilder.java

```diff
@@ -251,29 +247,26 @@ public AzureMonitorMetricExporter buildMetricExporter() {
     }
 
     /**
-     * Creates an {@link AzureMonitorLogExporter} based on the options set in the builder. This
-     * exporter is an implementation of OpenTelemetry {@link LogExporter}.
+     * Creates an {@link AzureMonitorLogRecordExporter} based on the options set in the builder. This
+     * exporter is an implementation of OpenTelemetry {@link LogRecordExporter}.
      *
-     * @return An instance of {@link AzureMonitorLogExporter}.
+     * @return An instance of {@link AzureMonitorLogRecordExporter}.
      * @throws NullPointerException if the connection string is not set on this builder or if the
      * environment variable "APPLICATIONINSIGHTS_CONNECTION_STRING" is not set.
      */
-    public AzureMonitorLogExporter buildLogExporter() {
-        return new AzureMonitorLogExporter(
-            new LogDataMapper(true, this::populateDefaults), initExporterBuilder());
+    public LogRecordExporter buildLogRecordExporter() {
+        return new AzureMonitorLogRecordExporter(
+            new LogDataMapper(true, false, this::populateDefaults), initExporterBuilder());
     }
 
     private TelemetryItemExporter initExporterBuilder() {
-        if (this.connectionString == null) {
-            // if connection string is not set, try loading from configuration
-            Configuration configuration = Configuration.getGlobalConfiguration();
-            connectionString(configuration.get(APPLICATIONINSIGHTS_CONNECTION_STRING));
+        String connectionStringOverride = configuration.get(APPLICATIONINSIGHTS_CONNECTION_STRING);
+        if (connectionStringOverride != null) {
+            // APPLICATIONINSIGHTS_CONNECTION_STRING environment variable should take precedence over
+            // programmatically calling connectionString()
```

Comment:
This intrigues me as the general pattern has been the inverse where environment configurations are overridden by programmatic choices. Is there a specific reason why this flips that?

Reply:
I double checked the expected behavior on our side, and it should be the way you mention 👍